apps/ensapi/src/lib/ensanalytics/referrer-leaderboard/database.ts

```diff
@@ -0,0 +1,72 @@
+import { and, count, desc, eq, gte, isNotNull, lte, ne, sql, sum } from "drizzle-orm";
+import { type Address, zeroAddress } from "viem";
+
+import * as schema from "@ensnode/ensnode-schema";
+import { type AccountId, serializeAccountId, type UnixTimestamp } from "@ensnode/ensnode-sdk";
+
+import { db } from "@/lib/db";
+
+export type ReferrerLeaderboardRecord = {
+  referrer: Address;
+
+  totalReferrals: number;
+
+  totalIncrementalDuration: string;
+};
+
+/**
+ * Get Referrer Leaderboard Records
```

Comment:
```suggestion
 * Get referrer leaderboard records from all database records in the `registrar_actions` table
 * matching the provided filters.
```

Goal: Transfering this idea out of `apps/ensapi/src/lib/ensanalytics/referrer-leaderboard/get-referrer-leaderboard.ts` which shouldn't talk about database internals.